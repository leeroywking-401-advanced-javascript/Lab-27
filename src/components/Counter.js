import React from "react";


class Counter extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
        counter: 0
      };
    }
  
    handleButtonClickUp = e => {
      e.preventDefault();
      this.setState({ counter: ++this.state.counter });
    };
  
    handleButtonClickDown = e => {
      e.preventDefault();
      this.setState({ counter: --this.state.counter });
    };
  
    render() {
      return (
        <div id="main">
          <button
            class="clicker"
            id="clickdown"
            onClick={this.handleButtonClickDown}
          >
            -1
          </button>
          <h4 id="counter">{this.state.counter}</h4>
          <button class="clicker" id="clickup" onClick={this.handleButtonClickUp}>
            +1
          </button>
        </div>
      );
    }
  }
  
  export default Counter