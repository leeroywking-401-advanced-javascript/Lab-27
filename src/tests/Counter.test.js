import React from 'react'
import renderer from 'react-test-renderer'

import Counter from '../components/Counter.js'
import { exportAllDeclaration, isTSAnyKeyword } from '@babel/types';

describe('<Counter />', () => {
  it('is alive at start', () => {
    let component = shallow(<Counter />);
    expect(component.exists()).toBeTruthy();
  });

  it('+1s on a click up', () => {
    let component = mount(<Counter />);
    let buttonUp = component.find('#clickup');
    buttonUp.simulate('click');
    expect(component.state('counter')).toBe(1);
  });

  it('-1s on a click down', () => {
    let component = mount(<Counter />);
    let buttondown = component.find('#clickdown');
    buttondown.simulate('click');
    expect(component.state('counter')).toBe(-1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Counter />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})