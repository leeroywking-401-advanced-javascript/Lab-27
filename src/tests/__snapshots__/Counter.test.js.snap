// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counter /> renders correctly 1`] = `
<div
  id="main"
>
  <button
    class="clicker"
    id="clickdown"
    onClick={[Function]}
  >
    -1
  </button>
  <h4
    id="counter"
  >
    0
  </h4>
  <button
    class="clicker"
    id="clickup"
    onClick={[Function]}
  >
    +1
  </button>
</div>
`;
